File: MotivApp/app/src/main/java/com/motiv/piotr/dao/LinkDao.java
package com.motiv.piotr.dao;

import androidx.fragment.app.*;
import androidx.room.*;
import com.motiv.piotr.Link;
import dagger.*;
import dagger.android.*;
import dagger.android.support.*;
import javax.inject.*;

@Dao
public interface LinkDao {

    @Insert
    public long saveLink(com.motiv.piotr.Link link);

    @Delete
    public int deleteLink(com.motiv.piotr.Link link);

    @Query("SELECT * FROM link")
    public java.util.List<Link> getLinks();

    @Query("SELECT * FROM link WHERE id = :id")
    public Link getLink(java.lang.String id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public void saveLinks(java.util.List<com.motiv.piotr.Link> link);
}

File: MotivApp/app/src/main/java/com/motiv/piotr/dao/LinkDao.kt
 
package com.motiv.piotr.dao
import androidx.fragment.app.*
import androidx.room.*
import com.motiv.piotr.Link
import dagger.*
import dagger.android.*
import dagger.android.support.*
import javax.inject.*
import kotlin.collections.List
@Dao
public interface LinkDao {

    @Insert
    public fun saveLink(link: com.motiv.piotr.Link): Long@Delete
    public fun deleteLink(link: com.motiv.piotr.Link): Int@Query("SELECT * FROM link")
    public fun getLinks(): List<Link>@Query("SELECT * FROM link WHERE id = :id")
    public fun getLink(id: String): Link@Insert(onConflict = OnConflictStrategy.REPLACE)
    public fun saveLinks(link: List<com.motiv.piotr.Link>): Unit
}
